src/binary-writer.cc

```diff
@@ -720,8 +720,15 @@ void BinaryWriter::WriteSimdLoadStoreLaneExpr(const Func* func,
   auto* typed_expr = cast<T>(expr);
   WriteOpcode(stream_, typed_expr->opcode);
   Address align = typed_expr->opcode.GetAlignment(typed_expr->align);
-  stream_->WriteU8(log2_u32(align), "alignment");
-  WriteU32Leb128(stream_, typed_expr->offset, desc);
+  Index memidx = module_->GetMemoryIndex(typed_expr->memidx);
+  if (memidx != 0) {
+    stream_->WriteU8(log2_u32(align) | (1 << 6), "alignment");
+    WriteU32Leb128(stream_, memidx, "memidx");
+    WriteU32Leb128(stream_, typed_expr->offset, desc);
+  } else {
+    stream_->WriteU8(log2_u32(align), "alignment");
+    WriteU32Leb128(stream_, typed_expr->offset, desc);
```

Comment:
Wove this last line outside the condition?

Reply:
Done.